### PR TITLE
[BUGFIX] Fix bug in Query template values metric when it gets int values

### LINKED ISSUE #8607: [BUG] Bug in Query template values metric when it gets int values
**Describe the bug**
Hi, 
The new version of GX with #8570  was supposed to fix a bug in Query template values metric, but [this addition](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/query_metrics/query_template_values.py#L30) introduced another bug. 

**To Reproduce**
Trying to run this expectation will result in an error:, because sa.text() doesnt like an Int value (which is provided in "num_of_distinct_values" inside the template_dict)

```
{
        "expectation_type": "expect_queried_column_to_have_n_distinct_values_with_condition",
        "kwargs": {"template_dict": {'condition': 'is_current', 'column_to_check': 'end_date',
                                     'num_of_distinct_values': 1
                                     },
                  }
    }
```

The error message is: TypeError: expected string or bytes-like object.
Stack trace: 
```
File "/Users/itaise/github/wix-data-dev/wix_venv/lib/python3.10/site-packages/great_expectations/expectations/metrics/query_metrics/query_template_values.py", line 30, in get_query
template_dict_reformatted = {
File "/Users/itaise/github/wix-data-dev/wix_venv/lib/python3.10/site-packages/great_expectations/expectations/metrics/query_metrics/query_template_values.py", line 31, in
k: sa.text(v).format(active_batch=selectable)
File "", line 2, in text
File "", line 2, in _create_text
File "/Users/itaise/github/wix-data-dev/wix_venv/lib/python3.10/site-packages/sqlalchemy/util/deprecations.py", line 375, in warned
return fn(*args, **kwargs)
File "/Users/itaise/github/wix-data-dev/wix_venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 1922, in _create_text
return TextClause(text, bind=bind)
File "/Users/itaise/github/wix-data-dev/wix_venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 1822, in __init__
self.text = self._bind_params_regex.sub(repl, text)
TypeError: expected string or bytes-like object
```

@Kilo59  Can you please check? thanks! 

**Environment (please complete the following information):**
 - Great Expectations Version: great-expectations==0.17.11

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What does this PR do? 
* Follow up to #8570 and #8562
* Fixes #8607

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/metrics/query_metrics/query_template_values.py [hunk 1]
```diff
@@ -28,8 +28,8 @@ class QueryTemplateValues(QueryMetricProvider):
     @classmethod
     def get_query(cls, query, template_dict, selectable):
         template_dict_reformatted = {
-            k: sa.text(v).format(active_batch=selectable)
-            if isinstance(v, int)
+            k: str(v).format(active_batch=selectable)
+            if isinstance(v, (float, int))
```
Comment (Kilo59) [inline great_expectations/expectations/metrics/query_metrics/query_template_values.py]:
```suggestion
            if isinstance(v, float)
```
A an `int` is a subclass of `float`, so just checking for the `float` should be enough.

**Review by Kilo59 — COMMENTED** [review]
@Shinnnyshinshin is there a unit-test that could catch a regression here? One that would fail without this fix.

File: great_expectations/expectations/metrics/query_metrics/query_template_values.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/expectations/metrics/query_metrics/query_template_values.py]: Tried this, but it doesn't work, unfortunately. It goes to the `else` branch, and gives the following error. 

```
E   AttributeError: 'int' object has no attribute 'format'
```

**Comment by Shinnnyshinshin** [comment]
> @Shinnnyshinshin is there a unit-test that could catch a regression here? One that would fail without this fix.

@Kilo59  - Test has been added

File: great_expectations/expectations/metrics/query_metrics/query_template_values.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/expectations/metrics/query_metrics/query_template_values.py]: That doesn't seem right based on this rule, maybe this was changed in a later version of python (or the rule is just wrong).
https://beta.ruff.rs/docs/rules/redundant-numeric-union/

You could do this...
```python
import numbers

if isinstance(v, number.Number)
```


https://docs.python.org/3/library/numbers.html#numbers.Number

**Review by Kilo59 — APPROVED** [review]

File: great_expectations/expectations/metrics/query_metrics/query_template_values.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/expectations/metrics/query_metrics/query_template_values.py]: oo sure. I can add this

**Pull request merged**